app/client/keybase/keyvault_test.go

```diff
@@ -0,0 +1,192 @@
+package keybase
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	vault "github.com/hashicorp/vault/api"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func setupTestVaultKeybase(address string) (*vault.Client, *VaultKeybase, error) {
+	config := vault.DefaultConfig()
+
+	config.Address = address
+
+	// Create a new Vault API client
+	client, err := vault.NewClient(config)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Set the root token for the client
+	client.SetToken("dev-only-token")
+
+	// Create a new VaultKeybase instance
+	vk := NewVaultKeybase(client, "secret")
+
+	return client, vk, nil
+}
+
+func TestVaultKeybase(t *testing.T) {
```

Comment:
Consider using `TestMain` for the setup and teardown of the vault container: https://pkg.go.dev/testing.

We do something similar with our postgres container in `persistence/test/setup_test.go`